File: src/main/java/api/TinyResponse.java
package api;

public class TinyResponse {
    public Input input;
    public Output output;

    public class Input {
        public int size;
        public String type;
    }

    public class Output {
        public int size;
        public String type;
        public int width;
        public int height;
        public double ratio;
        public String url;
    }
}

File: src/main/kotlin/api/TinyResponse.kt
package api

class TinyResponse {
    lateinit var input: Input
    lateinit var output: Output

    inner class Input {
        var size: Int = 0
        lateinit var type: String
    }

    inner class Output {
        var size: Int = 0
        lateinit var type: String
        var width: Int = 0
        var height: Int = 0
        var ratio: Double = 0.toDouble()
        lateinit var url: String
    }

    override fun equals(other: Any?): Boolean {
        return other is TinyResponse
                && this.input.size == other.input.size
                && this.output.size == other.output.size
    }
}